fennel/sources/sources.py

```diff
@@ -56,9 +57,16 @@ def source(
     if since is not None and not isinstance(since, datetime):
         raise TypeError(f"'since' must be of type datetime - got {type(since)}")
 
+    if until is not None and not isinstance(until, datetime):
+        raise TypeError(f"'until' must be of type datetime - got {type(until)}")
+
+    if since is not None and until is not None and since > until:
+        raise ValueError("'since' must be before 'until'")
```

Comment:
maybe `earlier than` instead of `before`? Makes the fields in the error message feel more time dimensional based.